[FLINK-31050] Add cases for IN and NOT IN predicate in ORC format

SteNicholas opened the pull request:
Supports `IN` and `NOT IN` predicate push down in ORC format for `OrcPredicateFunctionVisitor`. In the `IN` predicate, 20 literals are critical for performance. If there are more than 20 literals, the performance will decrease.

**The brief change log**

- Supports `visitIn` and `visitNotIn` in `OrcPredicateFunctionVisitor`.

FangYongs commented on flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/filter/OrcPredicateFunctionVisitor.java:
```diff
@@ -157,20 +158,34 @@ private Optional<OrcFilters.Predicate> convertBinary(
+                                            // validate that literal is serializable
+                                            return orcObj instanceof Serializable
+                                                    ? (Serializable) orcObj
+                                                    : null;
```
Should we put `null` in literals results or return `Optional.empty` for `convertIn`?

SteNicholas replied:
@FangYongs, IMO, it's better to filter serializable literal for the literals, and don't care the non-serializable literal. WDYT?

SteNicholas replied:
Make sense to me. Keeps the consistence with the non-serializable object of `convertBinary`.

FangYongs commented:
Thanks @SteNicholas , LGTM

JingsongLi reviewed (commented):
I am worried that too large values in `IN` will lead to poor performance. For example, in the case of 100 values, each execution may require 100 evals, the execution efficiency is very poor.

SteNicholas commented:
@JingsongLi, @FangYongs, according to above comment, I have removed the `Between` and `In` of `OrcFilters` and added the comment for 20 literals performance critical. PTAL.

JingsongLi reviewed (approved):
+1
